### ISSUE TITLE
Make sure the library works with ES-modules

### ISSUE THREAD

**Issue by molefrog** [molefrog/wouter#10]
As @ai pointed out on Twitter, things `from './history'` won't work in native ES modules. Need to simulate this and make sure everything is written according to the spec.

**Comment by ai** [comment]
The main problem, that there is not good production solution for ES modules to test how it will work in real ES modules in wildlife :(.

ES modules for browsers requires a new generation of bundlers (to fix paths and make preload maps). We still need some features (like import maps) which didn’t ship yet. `pike` is a great start, but it could change API a few times before the final release.

**Comment by molefrog** [comment]
Right, makes absolutely sense. One of problems is being able to import things from `node_modules`.

I made a small local demo with react being replaced with a global variable. You were right, the current code does not work, since the bare import specifiers aren't supported by the browsers. However, when I changed it to `import .. from "./history.js"` it fixed the problem:
<img width="907" alt="Screenshot 2019-04-27 at 21 06 20" src="https://user-images.githubusercontent.com/671276/56853526-e18dee80-6931-11e9-9dec-60120cdf121c.png">
*(it worked in Chrome, Safari and Edge, but not in IE11 obviously, since the code was untranspiled)*

So I also decided to make a thorough inspection of the code to make sure I only use ES2015 (ES6) features. Here is full list of things the library uses:
  - arrow function
  - destructing assignment
  - default arguments & arguments spread
  - let / const

That's it! We should be good now.
